src/coprocessor/readpool_impl.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::cell::RefCell;
+
+use crate::pd::PdTask;
+use crate::server::readpool::{self, Builder, ReadPool};
+use crate::util::collections::HashMap;
+use crate::util::worker::FutureScheduler;
+
+use super::metrics::*;
+use prometheus::local::*;
+
+use crate::coprocessor::dag::executor::ExecutorMetrics;
+
+thread_local! {
+    pub static LOCAL_COPR_REQ_HISTOGRAM_VEC: RefCell<LocalHistogramVec> =
```

Comment:
oh, we have so many metrics, it is better to use a structure to wrap all so we can only use one thread local var instead? /cc @breeswish

Reply:
I'm fine with both, maybe not much difference.

Reply:
em, maybe we can do a benchmark, one local struct vs multi local vars

Reply:
ok, I will do a benchmark about this.